gitlab-container-registry-4.17.1: fails tests

### Nixpkgs version

- Stable (24.11)

### Describe the bug

gitlab-container-registry fails its tests. I noticed this when updating Gitlab to 17.9.2.

### Steps to reproduce

Set
```
services.gitlab.registry = {
  enable = true;
  package = pkgs.gitlab-container-registry;
};
```
and rebuild the system.

### Expected behaviour

Tests should pass

### Screenshots

_No response_

### Relevant log output

```console
error: builder for '/nix/store/5w2ccbm4ywjk66ql3an4hvchhp4prnz5-gitlab-container-registry-4.17.1.drv' failed with exit code 1;
       last 25 log lines:
       >               Error Trace:    /build/source/registry/storage/driver/s3-aws/s3_test.go:398
       >                    Error:          Received unexpected error:
       >                                     s3aws: NoCredentialProviders: no valid providers in chain. Deprecated.
       >                                         For verbose messaging see aws.Config.CredentialsChainVerboseErrors
       >                     Test:           TestS3DriverPathStyle/path_style_disabled_with_custom_endpoint
       >                 Messages:       unable to generate URL
       >     --- FAIL: TestS3DriverPathStyle/path_style_enabled_with_custom_endpoint (1.68s)
       >         s3_test.go:398:
       >                   Error Trace:    /build/source/registry/storage/driver/s3-aws/s3_test.go:398
       >                    Error:          Received unexpected error:
       >                                     s3aws: NoCredentialProviders: no valid providers in chain. Deprecated.
       >                                         For verbose messaging see aws.Config.CredentialsChainVerboseErrors
       >                     Test:           TestS3DriverPathStyle/path_style_enabled_with_custom_endpoint
       >                  Messages:       unable to generate URL
       >     --- FAIL: TestS3DriverPathStyle/default_path_style_with_custom_endpoint (1.60s)
       >         s3_test.go:398:
       >                   Error Trace:    /build/source/registry/storage/driver/s3-aws/s3_test.go:398
       >                    Error:          Received unexpected error:
       >                                     s3aws: NoCredentialProviders: no valid providers in chain. Deprecated.
       >                                         For verbose messaging see aws.Config.CredentialsChainVerboseErrors
       >                     Test:           TestS3DriverPathStyle/default_path_style_with_custom_endpoint
       >                  Messages:       unable to generate URL
       > FAIL
       > FAIL      github.com/docker/distribution/registry/storage/driver/s3-aws   8.176s
       > FAIL
       For full logs, run 'nix log /nix/store/5w2ccbm4ywjk66ql3an4hvchhp4prnz5-gitlab-container-registry-4.17.1.drv'.
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.80, NixOS, 24.11 (Vicuna), 24.11.20250304.d29772b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - channels(root): `""`
 - nixpkgs: `/nix/store/42k3knjp8qlqaa2mxaqhsh6mv421v32n-source`

### Notify maintainers

@yayayayaka 
@xanderio
@blitz 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/gi/gitlab-container-registry/package.nix
+++ b/pkgs/by-name/gi/gitlab-container-registry/package.nix
@@ -20,12 +20,11 @@ buildGoModule rec {
 
   vendorHash = "sha256-oNQoKn8GPJxmUzkUHGzax2/KWyI3VXLRtAvWe9B64Ds=";
 
-  postPatch = ''
-    substituteInPlace health/checks/checks_test.go \
-      --replace-fail \
-        'func TestHTTPChecker(t *testing.T) {' \
-        'func TestHTTPChecker(t *testing.T) { t.Skip("Test requires network connection")'
-  '';
+  checkFlags = [
+    # TestHTTPChecker requires internet
+    # TestS3DriverPathStyle requires s3 credentials/urls
+    "-skip TestHTTPChecker|TestS3DriverPathStyle"
+  ];
 
   meta = with lib; {
     description = "GitLab Docker toolset to pack, ship, store, and deliver content";

